### ISSUE TITLE
Build fails on OS X with Erlang R17

### ISSUE THREAD

**Issue by aeden** [basho/riak#695]
I am attempting to build Riak on OS X 10.10 with R17.

Erlang/OTP 17 [erts-6.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Here is the output of the build:

```
./rebar get-deps
==> rel (get-deps)
==> riak (get-deps)
==> riak (get-deps)
Pulling node_package from {git,"git://github.com/basho/node_package.git",
                               {branch,"develop"}}
Cloning into 'node_package'...
Pulling lager_syslog from {git,"git://github.com/basho/lager_syslog.git",
                               {tag,"2.0.3"}}
Cloning into 'lager_syslog'...
Pulling cluster_info from {git,"git://github.com/basho/cluster_info.git",
                               {branch,"develop"}}
Cloning into 'cluster_info'...
Pulling riak_kv from {git,"git://github.com/basho/riak_kv.git",
                          {branch,"develop"}}
Cloning into 'riak_kv'...
Pulling riak_search from {git,"git://github.com/basho/riak_search.git",
                              {branch,"develop"}}
Cloning into 'riak_search'...
Pulling riak_control from {git,"git://github.com/basho/riak_control.git",
                               {branch,"develop"}}
Cloning into 'riak_control'...
Pulling riaknostic from {git,"git://github.com/basho/riaknostic.git",
                             {branch,"develop"}}
Cloning into 'riaknostic'...
Pulling yokozuna from {git,"git://github.com/basho/yokozuna.git",
                           {branch,"develop"}}
Cloning into 'yokozuna'...
Pulling riak_auth_mods from {git,"git://github.com/basho/riak_auth_mods.git",
                                 {branch,"develop"}}
Cloning into 'riak_auth_mods'...
Pulling rebar_lock_deps_plugin from {git,
                                     "git://github.com/seth/rebar_lock_deps_plugin.git",
                                     {branch,"master"}}
Cloning into 'rebar_lock_deps_plugin'...
Pulling meck from {git,"git://github.com/basho/meck.git",{tag,"0.8.2"}}
Cloning into 'meck'...
==> node_package (get-deps)
==> lager_syslog (get-deps)
Pulling lager from {git,"git://github.com/basho/lager.git",{tag,"2.0.3"}}
Cloning into 'lager'...
Pulling syslog from {git,"git://github.com/Vagabond/erlang-syslog.git",
                         {tag,"1.0.2"}}
Cloning into 'syslog'...
==> lager (get-deps)
Pulling goldrush from {git,"git://github.com/DeadZen/goldrush.git",
                           {tag,"0.1.6"}}
Cloning into 'goldrush'...
==> goldrush (get-deps)
==> syslog (get-deps)
==> cluster_info (get-deps)
==> riak_kv (get-deps)
Pulling sidejob from {git,"git://github.com/basho/sidejob.git",
                          {branch,"develop"}}
Cloning into 'sidejob'...
Pulling erlang_js from {git,"git://github.com/basho/erlang_js.git",
                            {tag,"1.3.0"}}
Cloning into 'erlang_js'...
Pulling bitcask from {git,"git://github.com/basho/bitcask.git",
                          {branch,"develop"}}
Cloning into 'bitcask'...
Pulling eper from {git,"git://github.com/basho/eper.git",{tag,"0.78"}}
Cloning into 'eper'...
Pulling sext from {git,"git://github.com/basho/sext.git",{tag,"1.1p3"}}
Cloning into 'sext'...
Pulling riak_pipe from {git,"git://github.com/basho/riak_pipe.git",
                            {branch,"develop"}}
Cloning into 'riak_pipe'...
Pulling riak_api from {git,"git://github.com/basho/riak_api.git",
                           {branch,"develop"}}
Cloning into 'riak_api'...
Pulling riak_dt from {git,"git://github.com/basho/riak_dt.git",
                          {branch,"develop"}}
Cloning into 'riak_dt'...
Pulling eunit_formatters from {git,"git://github.com/seancribbs/eunit_formatters",
                                   {tag,"0.1.2"}}
Cloning into 'eunit_formatters'...
==> sidejob (get-deps)
==> erlang_js (get-deps)
==> meck (get-deps)
==> bitcask (get-deps)
Pulling cuttlefish from {git,"git://github.com/basho/cuttlefish.git",
                             {branch,"develop"}}
Cloning into 'cuttlefish'...
==> cuttlefish (get-deps)
Pulling getopt from {git,"git://github.com/jcomellas/getopt.git",
                         {tag,"v0.4.3"}}
Cloning into 'getopt'...
Pulling neotoma from {git,"git://github.com/seancribbs/neotoma.git",
                          {tag,"1.7.2"}}
Cloning into 'neotoma'...
==> getopt (get-deps)
==> neotoma (get-deps)
==> eper (get-deps)
==> sext (get-deps)
Pulling edown from {git,"git://github.com/uwiger/edown.git",{tag,"0.5"}}
Cloning into 'edown'...
==> edown (get-deps)
==> riak_pipe (get-deps)
Pulling riak_core from {git,"git://github.com/basho/riak_core.git",
                            {branch,"develop"}}
Cloning into 'riak_core'...
==> riak_core (get-deps)
Pulling poolboy from {git,"git://github.com/basho/poolboy.git",
                          {tag,"0.8.1p3"}}
Cloning into 'poolboy'...
Pulling basho_stats from {git,"git://github.com/basho/basho_stats.git",
                              {tag,"1.0.3"}}
Cloning into 'basho_stats'...
Pulling riak_sysmon from {git,"git://github.com/basho/riak_sysmon.git",
                              {branch,"develop"}}
Cloning into 'riak_sysmon'...
Pulling riak_ensemble from {git,"git://github.com/basho/riak_ensemble",
                                {branch,"develop"}}
Cloning into 'riak_ensemble'...
Pulling pbkdf2 from {git,"git://github.com/basho/erlang-pbkdf2.git",
                         {tag,"2.0.0"}}
Cloning into 'pbkdf2'...
Pulling eleveldb from {git,"git://github.com/basho/eleveldb.git",
                           {branch,"develop"}}
Cloning into 'eleveldb'...
Pulling exometer_core from {git,"git://github.com/basho/exometer_core.git",
                                {tag,"1.0.0-basho2"}}
Cloning into 'exometer_core'...
Pulling clique from {git,"git://github.com/basho/clique.git",{tag,"0.2.5"}}
Cloning into 'clique'...
==> poolboy (get-deps)
==> basho_stats (get-deps)
==> riak_sysmon (get-deps)
==> eleveldb (get-deps)
Cloning into 'leveldb'...
Note: checking out '2.0.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at d1a95db... Merge pull request #142 from basho/mv-comparator-segfault
==> riak_ensemble (get-deps)
==> pbkdf2 (get-deps)
==> src (get-deps)
==> exometer_core (get-deps)
Pulling parse_trans from {git,"git://github.com/uwiger/parse_trans.git",
                              {tag,"2.9"}}
Cloning into 'parse_trans'...
Pulling folsom from {git,"git://github.com/basho/folsom.git",{tag,"0.7.4p5"}}
Cloning into 'folsom'...
Pulling setup from {git,"git://github.com/uwiger/setup.git",{tag,"1.4"}}
Cloning into 'setup'...
==> parse_trans (get-deps)
==> folsom (get-deps)
Pulling bear from {git,"git://github.com/basho/bear.git",{tag,"0.1.3p1"}}
Cloning into 'bear'...
==> bear (get-deps)
==> setup (get-deps)
==> clique (get-deps)
==> riak_api (get-deps)
Pulling riak_pb from {git,"git://github.com/basho/riak_pb.git",
                          {tag,"2.0.0.17"}}
Cloning into 'riak_pb'...
Pulling webmachine from {git,"git://github.com/basho/webmachine.git",
                             {tag,"1.10.8"}}
Cloning into 'webmachine'...
==> riak_pb (get-deps)
Pulling protobuffs from {git,"git://github.com/basho/erlang_protobuffs.git",
                             {tag,"0.8.1p5"}}
Cloning into 'protobuffs'...
==> protobuffs (get-deps)
==> webmachine (get-deps)
Pulling mochiweb from {git,"git://github.com/basho/mochiweb.git",
                           {tag,"v2.9.0p1"}}
Cloning into 'mochiweb'...
==> mochiweb (get-deps)
==> riak_dt (get-deps)
==> eunit_formatters (get-deps)
==> lucene_parser (get-deps)
==> riak_search (get-deps)
Pulling merge_index from {git,"git://github.com/basho/merge_index.git",
                              {branch,"develop"}}
Cloning into 'merge_index'...
==> merge_index (get-deps)
==> riak_control (get-deps)
Pulling erlydtl from {git,"git://github.com/evanmiller/erlydtl.git",
                          "d20b53f0"}
Cloning into 'erlydtl'...
==> erlydtl (get-deps)
==> riaknostic (get-deps)
==> yokozuna (get-deps)
Pulling kvc from {git,"git://github.com/etrepum/kvc.git",{tag,"v1.5.0"}}
Cloning into 'kvc'...
Pulling ibrowse from {git,"git://github.com/cmullaparthi/ibrowse.git",
                          {tag,"v4.0.2"}}
Cloning into 'ibrowse'...
==> kvc (get-deps)
==> ibrowse (get-deps)
==> riak_auth_mods (get-deps)
Pulling canola from {git,"git://github.com/basho/canola.git",{tag,"2.0.0"}}
Cloning into 'canola'...
==> canola (get-deps)
==> rebar_lock_deps_plugin (get-deps)
./rebar compile
==> node_package (compile)
ERROR: OTP release 17 does not match required regex R16
ERROR: compile failed while processing /Users/aeden/Development/basho/riak/deps/node_package: rebar_abort
make: *** [compile] Error 1
```

**Comment by cmeiklejohn** [comment]
We're in the progress of addressing these issues, one component at a time.  The next major release of Riak will be targeted at Erlang 17, but right now, only R16B02 is officially supported.

**Comment by aeden** [comment]
Oh, ok, thanks. Maybe update the README indicating that it can only be built with version up to R16B02.

**Comment by cmeiklejohn** [comment]
Opened #697 to address.

**Comment by cmeiklejohn** [comment]
Addressed by #699 (and a slew of other PRs across several repos.)

**Comment by aeden** [comment]
Thanks, I've confirmed the build succeeds now on my machine.
